### ISSUE TITLE
Make API calls easier

### ISSUE THREAD

**Issue by 304NotModified** [NLog/NLog#1169]
As discussed in https://github.com/NLog/NLog/issues/1164

Proposal:
1. Add maxLevel to contructor: 
   
   ``` cs
   public LoggingRule(string loggerNamePattern, 
                               LogLevel minLevel, 
                               LogLevel maxLevel, 
                              Target target)
   ```
2. create an `LoggingConfiguration.AddRule` and `LoggingConfiguration.RemoveRule`, analogous to the current `LoggingConfiguration.AddTarget` and `LoggingConfiguration.RemoveTarget` (targetname should be a `string`?)

Example

``` c#

var generalFileTarget = new FileTarget();
config.AddTarget("general", generalFileTarget);
config.AddRule("*",  LogLevel.Debug, generalFileTarget); //target type
config.AddRule("*",  LogLevel.Error, "target2"); // target name

```
1.  `RegisterDefinition` without name (use the attribute)  + generic version

now:

``` c#
//target
ConfigurationItemFactory.Default.Targets
    .RegisterDefinition("MyFirst", typeof(MyNamespace.MyFirstTarget));

    //layout renderer
ConfigurationItemFactory.Default.LayoutRenderers
    .RegisterDefinition("hello-world", typeof(MyNamespace.HelloWorldLayoutRenderer ));

```

proposal:

``` c#

//target
ConfigurationItemFactory.Default.Targets
    .RegisterDefinition<MyNamespace.MyFirstTarget>();

//layout renderer
ConfigurationItemFactory.Default.LayoutRenderers
    .RegisterDefinition<MyNamespace.HelloWorldLayoutRenderer>();

```

**Comment by snakefoot** [comment]
Maybe resolved by  #3686 (NLog 4.7) and #4323 (NLog 5.0)  where once can do this:

```c#
LogManager.Setup().
   SetupExtensions(eb => {
          eb.AutoLoadAssemblies(false);
          eb.RegisterTarget<MyTarget>("MyTarget");
   });
```

**Comment by snakefoot** [comment]
Closing as resolved with the new NLog 5.0 fluent-API with `LogManager.Setup()`
